decentraland/unity-renderer
# refactor: move UUID components to new assembly

## What does this PR change?

This PR moves all the UUIDComponent related code to new assembly. 
It also wraps up the UUIDComponent with related systems into a new plugin that has its own lifecycle.

Logic should be almost intact, most of the changes are related with moving code.

The new lifecycle for the components can be observed in [UUIDEventsPlugin](https://github.com/decentraland/unity-renderer/blob/5ce051aa66892d386ec2d67a67710b5dc0f1b2e7/unity-renderer/Assets/DCLPlugins/UUIDEventsPlugin/UUIDEventsPlugin.cs#L15) file.

Note that now, when testing UUID events, you have to instance the plugin by just doing `new UUIDEventsPlugin()` and then dispose it by calling its `Dispose()` method. If you don't do this `ParcelScene` is not going to recognize the component id (because its registered to the factory by the plugin).

Systems affected are:
* **InputController_Legacy**: singleton removed, now its just created by the plugin and injected into other systems via DI. Update handled via UpdateEventHandler.
* **InteractionHoverCanvasController**: singleton removed, DI into other systems, reacts to `DataStore_Cursor` instead of being used directly.
* **PointerEventsController**: service removed, now its just created by the plugin. Update handled via UpdateEventHandler.
* **CursorController**: singleton removed, reacts to `DataStore_Cursor` instead of being used directly.
* All **UUIDComponents**: logic not changed at all, just moved.

Diff is messy because files were moved to the plugin folder. So I recommend seeking the `UUIDEventsPlugin` class, `CursorController` and `InteractionHoverCanvasController` to review this PR.

## How to test the changes?

Use: https://play.decentraland.zone/?renderer-branch=refactor/uuid-components-to-new-assembly

The UUIDComponents and the hover feedback should work as intended.

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
